File: src/test/java/com/nowsprinting/intellij_mob/config/MobProjectSettingsTest.java
package com.nowsprinting.intellij_mob.config;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MobProjectSettingsTest {

    @Test
    void noStateLoaded_setDEfault() {
        MobProjectSettings sut = new MobProjectSettings();
        sut.noStateLoaded();

        assertEquals(sut.wipBranch, "mob-session");
        assertEquals(sut.baseBranch, "master");
        assertEquals(sut.remoteName, "origin");
        assertFalse(sut.debug);
        assertEquals(sut.timer, 10);
        assertFalse(sut.startWithShare);
        assertFalse(sut.nextAtExpire);
        assertEquals(sut.wipCommitMessage, "mob next [ci-skip]");
        assertFalse(sut.nextStay);
    }
}

File: src/test/kotlin/com/nowsprinting/intellij_mob/config/MobProjectSettingsTest.kt
/*
 * Copyright 2020-2021 Koji Hasegawa. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
 */

package com.nowsprinting.intellij_mob.config

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class MobProjectSettingsTest {

    @Test
    fun noStateLoaded_setDefault() {
        val sut = MobProjectSettings()
        sut.noStateLoaded()
        Assertions.assertEquals(sut.wipBranch, "mob-session")
        Assertions.assertEquals(sut.baseBranch, "master")
        Assertions.assertEquals(sut.remoteName, "origin")
        Assertions.assertEquals(sut.timerMinutes, 10)
        Assertions.assertFalse(sut.startWithShare)
        Assertions.assertEquals(sut.wipCommitMessage, "mob next [ci-skip]")
        Assertions.assertTrue(sut.nextStay)
    }
}